docs/refman/loaderbase.py

```diff
@@ -73,6 +73,13 @@ def _resolve_type(self, raw: str) -> Type:
             typ.resolved += [DataTypeInfo(obj, held_type)]
         return typ
 
+    def _resolved_returned_by(self, resolved: T.List[DataTypeInfo], func: T.Union[Function, Method]) -> None:
+        for obj in resolved:
+            if func not in obj.data_type.returned_by:
+                obj.data_type.returned_by += [func]
+            if obj.holds:
+                self._resolved_returned_by(obj.holds.resolved, func)
```

Comment:
The `if func not in obj.data_type.returned_by` test is new. Does it solve a problem?

I think adding the assignment to this method is unnecessary. Why not flatten the resolved data type info and iterate over that?
```suggestion
    @classmethod
    def _flatten_resolved(cls, resolved: T.List[DataTypeInfo]) -> T.Generator(Object, None, None):
        for obj in resolved:
            yield obj.data_type
            if obj.holds:
                yield from cls._all_resolved(obj.holds.resolved)
```

Reply:
The `if func not in` test is a new problem. Because a function can return `str | list[str]`, it needs to be filtered so it doesn't show up as being returned twice.

Flattening the DataTypeInfo with a generator is a reasonable approach and keeps the diff closer to the original code. Applied with the change to return DataTypeInfo instead of Object, to minimize the diff.